File: DarkOrbit 10.0/Net/netty/commands/KillScreenPostCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class KillScreenPostCommand
    {
        public const short ID = 10114;

        public static byte[] write(string pKillerName, string pKillerEpppLink, string pPlayerShipAlias,
                                 DestructionTypeModule pCause, List<KillScreenOptionModule> pOptions)
        {
            var param1 = new ByteArray(ID);
            param1.writeInt(pOptions.Count);
            foreach(var opt in pOptions)
            {
                param1.write(opt.write());
            }
            param1.writeUTF(pKillerEpppLink);
            param1.write(pCause.write());
            param1.writeUTF(pPlayerShipAlias);
            param1.writeUTF(pKillerName);
            param1.writeShort(7114);
            return param1.ToByteArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/KillScreenPostCommand.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;
import java.util.Vector;

import simulator.netty.ServerCommand;

/**
 Created by Shock & Pedro on 30-03-2015.
 */
public class KillScreenPostCommand
        implements ServerCommand {

    public static int ID = 7217;

    public String                         mKillerName;
    public String                         mKillerEpppLink;
    public String                         mPlayerShipAlias;
    public DestructionTypeModule          mCause;
    public Vector<KillScreenOptionModule> mOptions;

    public KillScreenPostCommand(String pKillerName, String pKillerEpppLink, String pPlayerShipAlias,
                                 DestructionTypeModule pCause, Vector<KillScreenOptionModule> pOptions) {
        this.mKillerName = pKillerName;
        this.mKillerEpppLink = pKillerEpppLink;
        this.mPlayerShipAlias = pPlayerShipAlias;
        this.mCause = pCause;
        this.mOptions = pOptions;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeUTF(this.mPlayerShipAlias);
            param1.writeUTF(this.mKillerName);
            this.mCause.write(param1);
            param1.writeInt(this.mOptions.size());
            for (KillScreenOptionModule c : this.mOptions) {
                c.write(param1);
            }
            param1.writeUTF(this.mKillerEpppLink);
        } catch (IOException e) {
        }
    }
}
